tests/test_package.py

```diff
@@ -0,0 +1,68 @@
+import io
+import logging
+import os
+import shutil
+from contextlib import contextmanager
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from snowcli.cli.snowpark import package
+
+from tests.test_data import test_data
+
+
+class TestPackage:
+    LOGGER = logging.getLogger(__name__)
+
+    @pytest.mark.parametrize(
+        "argument",
+        [
+            (
+                "snowflake-connector-python",
+                "Package snowflake-connector-python is available on the Snowflake anaconda channel.",
+                "snowcli.cli.snowpark.package",
+            ),
+            (
+                "some-weird-package-we-dont-know",
+                "not found in Snowflake anaconda channel...",
+                "snowcli.utils",
+            ),
+        ],
+    )
+    @patch("tests.test_package.package.utils.requests")
+    def test_package_lookup(self, mock_requests, caplog, argument, monkeypatch):
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = test_data.anaconda_response
+        mock_requests.get.return_value = mock_response
+
+        monkeypatch.setattr("sys.stdin", io.StringIO("N"))
+
+        with caplog.at_level(logging.DEBUG, logger=argument[2]):
+            result = package.package_lookup(argument[0], install_packages=True)
+
+        assert caplog.text
+        assert argument[1] in caplog.text
```

Comment:
assert logs is the only way to test it?

Reply:
It felt, that functions in package.py rely on so many outside connections (anaconda, pip, etc.), that it makes more sense to test them in integration tests. But i'm open to new test case propositions.

Reply:
Don't see other way ☹️